### PR TITLE
feat(sdk): use public-facing question type in event handlers

### LINKED ISSUE #50487: Expose public-facing question type and transforms for sdk
Exposes a minimal, public-facing `MetabaseQuestion` type and transforms for use in the sdk.

### LINKED ISSUE #50505: Use public-facing question type across the sdk

### PR DESCRIPTION
**Pull request by heypoom**
Closes https://github.com/metabase/metabase/issues/50487, Closes https://github.com/metabase/metabase/issues/50505

### Description

Introduces the public-facing `MetabaseQuestion` type with transforms, and updates the existing event handlers to use the public metabase question type.

### How to verify

- Renders the Question and click on "Save"
- The `onSave` and `onBeforeSave` handlers should now give you the `MetabaseQuestion` type, instead of the internal type.

### Checklist

- [x] Tests have been added/updated to cover changes in this PR - Unit and E2E

### CODE REVIEW

**Review by WiNloSt — APPROVED** [review]
Looks good to me

File: enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/context/InteractiveQuestionProvider.tsx [hunk 1]
```diff
@@ -65,22 +66,24 @@ export const InteractiveQuestionProvider = ({
   const handleSave = async (question: Question) => {
     if (isSaveEnabled) {
       const saveContext = { isNewQuestion: false };
+      const publicQuestion = transformSdkQuestion(question);
```
[the remark this answers is not in the document]
Comment (heypoom) [reply enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/context/InteractiveQuestionProvider.tsx]: @WiNloSt Good call, I updated the name to `sdkQuestion` as that's clearer and the function is already called `transformSdkQuestion`

File: enterprise/frontend/src/embedding-sdk/types/public/question.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export interface MetabaseQuestion {
+  id: number;
+  name: string | null;
```
[the remark this answers is not in the document]
Comment (heypoom) [reply enterprise/frontend/src/embedding-sdk/types/public/question.ts:3]: @WiNloSt I just checked why `displayName` can be null and it seems like that only happens if card is undefined, which is impossible in the SDK's context. I'll remove `null` from the type and add a empty fallback then.

```tsx
  displayName(): string | null | undefined {
    return this._card && this._card.name;
  }
```

[the remark this answers is not in the document]
Comment (heypoom) [reply enterprise/frontend/src/embedding-sdk/types/public/question.ts]: Fixed in https://github.com/metabase/metabase/pull/50867/commits/4ff5aa53e7ab74a2184e9b4e8bc135696f798054

**Pull request merged**